ui/components/app/network-connection-banner/network-connection-banner.tsx

```diff
@@ -0,0 +1,185 @@
+  IconSize,
+} from '../../component-library';
+import {
+  BackgroundColor,
+  BlockSize,
+  IconColor,
+} from '../../../helpers/constants/design-system';
+import { useNetworkConnectionBanner } from '../../../hooks/useNetworkConnectionBanner';
+import { useTheme } from '../../../hooks/useTheme';
+import { NETWORKS_ROUTE } from '../../../helpers/constants/routes';
+import { setEditedNetwork } from '../../../store/actions';
+import { MetaMetricsEventName } from '../../../../shared/constants/metametrics';
+import { ThemeType } from '../../../../shared/constants/preferences';
+import { NetworkConnectionBanner as NetworkConnectionBannerType } from '../../../../shared/constants/app-state';
+
+type BannerIcon =
+  | {
+      type: 'spinner';
+      color: string;
+      verticalAdjustment: string;
+    }
+  | {
+      type: 'static';
+      color: IconColor;
+      name: IconName;
+      verticalAdjustment: string;
+    };
+
+const Spinner = ({
+  color,
+  size,
+  ...rest
+}: {
+  color: string;
+  size: number;
+} & SVGProps<SVGSVGElement>) => (
+  <svg
+    width={size}
+    height={size}
+    viewBox="0 0 24 24"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
+    {...rest}
+  >
+    {/* Draw an arc that covers 60% of the circle */}
+    <path
+      d="M 12 2 A 10 10 0 1 1 2 12"
+      stroke={color}
+      strokeWidth="2"
+      strokeLinecap="round"
+      fill="none"
+    >
+      <animateTransform
+        attributeName="transform"
+        type="rotate"
+        values="0 12 12;360 12 12"
+        dur="1.5s"
+        repeatCount="indefinite"
+      />
+    </path>
+  </svg>
+);
+
+const getBannerContent = (
+  networkConnectionBanner: Exclude<
+    NetworkConnectionBannerType,
+    { status: 'unknown' }
+  >,
+  theme: ThemeType,
+  t: ReturnType<typeof useI18nContext>,
+): {
+  message: string;
+  backgroundColor: BackgroundColor;
+  icon: BannerIcon;
+} => {
+  // Align the indicator with the text
+  const verticalAdjustment = '0.2rem';
+
+  if (networkConnectionBanner.status === 'degraded') {
+    const warningColor =
+      theme === 'light'
+        ? lightTheme.colors.warning.default
+        : darkTheme.colors.warning.default;
+    return {
+      message: t('stillConnectingTo', [networkConnectionBanner.networkName]),
+      backgroundColor: BackgroundColor.warningMuted,
+      icon: {
+        type: 'spinner',
+        color: warningColor,
+        verticalAdjustment,
+      },
+    };
+  }
+
+  return {
+    message: t('unableToConnectTo', [networkConnectionBanner.networkName]),
+    backgroundColor: BackgroundColor.errorMuted,
+    icon: {
+      type: 'static',
+      color: IconColor.errorDefault,
+      name: IconName.Danger,
+      verticalAdjustment,
+    },
+  };
+};
+
+export const NetworkConnectionBanner = () => {
+  const t = useI18nContext();
+  const theme = useTheme();
+  const navigate = useNavigate();
+  const dispatch = useDispatch();
+  const networkConnectionBanner = useNetworkConnectionBanner();
```

Comment:
Small: how about destructure `status` and `trackNetworkBannerEvent` here so that the `useCallback` dependencies is focused and not observing the entire object